src/helpers/daily-data-providers/combined-resting-heart-rate.ts

```diff
@@ -1,42 +1,42 @@
-﻿import { add, parseISO } from "date-fns";
-import { fitbitRestingHeartRateDataProvider } from ".";
+﻿import { add } from "date-fns";
+import { appleHealthRestingHeartRateDataProvider, fitbitRestingHeartRateDataProvider, garminRestingHeartRateDataProvider } from ".";
 import getDayKey from "../get-day-key";
-import queryAllDeviceData from "./query-all-device-data";
+import MyDataHelps from "@careevolution/mydatahelps-js";
 
 export default function (startDate: Date, endDate: Date) {
-	return queryAllDeviceData({
-		namespace: "AppleHealth",
-		type: "RestingHeartRate",
-		observedAfter: add(startDate, { days: -1 }).toISOString(),
-		observedBefore: add(endDate, { days: 1 }).toISOString()
-	}).then(function (ddp) {
-		return fitbitRestingHeartRateDataProvider(startDate, endDate).then(function (fitbitData) {
-			var dayValues: { [key: string]: number[] } = {};
-			ddp.forEach((d) => {
-				if (!d.startDate) { return; }
-				var day = getDayKey(parseISO(d.startDate));
-				var value = parseFloat(d.value);
-				if (!dayValues[day]) {
-					dayValues[day] = [];
-				}
-				dayValues[day].push(value);
-			});
+	let providers: Promise<{ [key: string]: number }>[] = [];
+
+	return MyDataHelps.getDataCollectionSettings().then((settings) => {
```

Comment:
Is there a way to elegantly fail the Promise on line 24 instead of checking the settings each time around?